core/include/irq.h

```diff
@@ -22,11 +22,14 @@
 #define IRQ_H
 
 #include <stdbool.h>
+#include "cpu_conf.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#ifdef LEGACY_IRQ_API
```

Comment:
I'm a bit late, but would like to propose two changes:

1. use IRQ_API_INLINED instead of LEGACY_IRQ_API

That turns around the logic, so only adapted architectures need to be changed. Less surprising to anyone having an arch outside riot.
Also, "LEGACY_IRQ_API" works only once, until we want to improve this again. then we'd have "EVEN_MORE_LEGACY_IRQ_API" ... So, instead give the define a name that describes what it does.


2. I think in this case defining "MAYBE_INLINE" to `static inline __attribute__((always_inline))`, if IRQ_API_INLINED is active, and to an empty define if not, saves us from duplicate declarations and doxygen.

Reply:
>  Less surprising to anyone having an arch outside riot.

Let us please not worry about out of tree archs. Maintaining and deprecating RIOTs public APIs is already enough pain in the ass and a significant toll in the development process; but for official APIs this is certainly justified. Worrying about breaking out-of-tree code that touches RIOT's internals is IMO really off limits.

Before the next release we should be done with the conversion. And IMO that is the point we should get rid of `IRQ_API_INLINED` or `LEGACY_IRQ_API`. So next release, out-of-tree archs would have to follow suit. Or better: Being PR'ed to get upstream and benefit from being maintained by the community.

Reply:
I personally don't mind either way, I was about to push the requested change. As I see it `LEGACY_IRQ_API` means touching more stuff now, but none at the end of the process, `IRQ_API_INLINED` means touching less stuff now, and all the platforms at the end. But if we actually change them all by the release `MAYBE_INLINE` would also be removed, having it as a place holder just reduces duplicated lines during migration. So IMO not much is gained or lost by any of the approaches.

Reply:
I pushed the requested change, I can either drop those or squash depending on what you prefer @kaspar030 @maribu

Reply:
I'm fine with either :-)

Reply:
Squashed.